test/dynamo/test_misc.py

```diff
@@ -1498,6 +1498,48 @@ def fn3(inputs):
         self.assertEqual(cnts.frame_count, 3)
         self.assertEqual(cnts.op_count, 9)
 
+    @patch.object(torch._dynamo.config, "capture_scalar_outputs", True)
+    def test_user_code_statically_known(self):
+        from torch.fx.experimental.symbolic_shapes import (
+            has_static_value,
+            statically_known_true,
+        )
+
+        @torch.compile(fullgraph=True)
+        def f(x):
+            # At this point, this isn't statically known, only the hint says so.
+            if statically_known_true(x.shape[0] > 9):
+                x -= 1
+            else:
+                x += 1
+            torch._check(x.shape[0] >= 10)
+            # But now it is.
+            if statically_known_true(x.shape[0] > 9):
+                x += 1
+            else:
+                x -= 1
+            if has_static_value(x.shape[0]):
+                x -= 1
+            else:
+                x += 1
+            return x
+
+        x = torch.zeros(10)
+        torch._dynamo.mark_dynamic(x, 0)
+        self.assertEqual(f(x).sum(), 30)
+
+        @torch.compile(fullgraph=True, dynamic=True)
+        def g(x, y):
+            n = x.item()
+            torch._check(n == 3)
+            if has_static_value(4.0) and has_static_value(n):
+                y += 1
+            return y
+
+        x = torch.tensor([3])
+        y = torch.zeros(1)
+        self.assertEqual(g(x, y).item(), 1)
```

Comment:
sorry this is a bit of mental gynmastics, can you just do:
```
return has_static_value(4.0), has_static_value(n)
```
and check the result is (True, False)?

Reply:
Ditto for the last test, I had to think about how it gets to 30